### PR TITLE
pkp/pkp-lib#7369 Multilingual text input field required validation

### PR DESCRIPTION
**Pull request by touhidurabir**
This PR aims to solve pkp/pkp-lib#7369

### CODE REVIEW

File: classes/form/Form.inc.php [hunk 1]
```diff
@@ -187,6 +188,7 @@ function fetch($request, $template = null, $display = false) {
 				'errors' => $this->getErrorsArray(),
 				'formLocales' => $this->supportedLocales,
 				'formLocale' => $this->getDefaultFormLocale(),
+				'primaryLocale' => $context ? $context->getPrimaryLocale() : $request->getSite()->getPrimaryLocale(),
```
[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/form/Form.inc.php:191]: @asmecher Can we set submission primary locale ? We can have different primary locale for the `Site` and `Context` and also have different locales for the submission from the journal/context `Settings Wizard`. But still the journal/context primary locale need to be required, right ? Just cant find option to set a submission primary locale, perhaps I am missing something ?

[the remark this answers is not in the document]
Comment (asmecher) [reply classes/form/Form.inc.php:191]: The submission's primary locale is set on the first page of the submission process, chosen by the author from among the set of supported submission locales that's defined in setup.

[the remark this answers is not in the document]
Comment (NateWr) [reply classes/form/Form.inc.php:193]: This PR looks good. @touhidurabir would you like me to pick it up, make this fix, and get it merged? Or do you want to commit a fix for me to review and merge?

[the remark this answers is not in the document]
Comment (NateWr) [reply classes/form/Form.inc.php:193]: Ah I see, great! This looks good. I haven't tested, but I think we also need the `primaryLocale` override set in the file upload form during the workflow (not the submission wizard). Here are reproduction steps for those forms:

1. Go to Submissions.
2. Click View on any submission in the Submission stage.
3. Find "Submission Files" and click "Upload" or "Add".
4. Upload a new file.
5. Enter a title for this file in the top locale but not the other locales.
6. While the title fields for all locales are still visible, try to click "Continue".
7. See the same error.

I think your fix here can be applied to that form to fix this. But if not let me know and we can address that separately.

[the remark this answers is not in the document]
Comment (touhidurabir) [reply classes/form/Form.inc.php:193]: @NateWr fixed and updated . was bit tricky to figure out where to make the change . just one question, what is the best/right alternative of `getLocal` method to get the submission primary locale as it's marked as deprecated in `3.2` ?

**Pull request merged**